internal/provider/resource_gitlab_personal_access_tokens.go

```diff
@@ -0,0 +1,242 @@
+	"api",
+	"read_user",
+	"read_api",
+	"read_repository",
+	"write_repository",
+	"read_registry",
+	"write_registry",
+	"sudo",
+}
+
+var _ = registerResource("gitlab_personal_access_token", func() *schema.Resource {
+	return &schema.Resource{
+		Description: `The ` + "`gitlab_personal_access_token`" + ` resource allows to manage the lifecycle of a personal access token for a specified user.
+
+-> **Note** In order to utilize this resource without failure the executing user must be an admin. 
+
+**Upstream API**: [GitLab REST API docs](https://docs.gitlab.com/ee/api/personal_access_tokens.html)`,
+
+		CreateContext: resourceGitlabPersonalAccessTokenCreate,
+		ReadContext:   resourceGitlabPersonalAccessTokenRead,
+		DeleteContext: resourceGitlabPersonalAccessTokenDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"user_id": {
+				Description: "The id of the user.",
+				Type:        schema.TypeInt,
+				ForceNew:    true,
+				Required:    true,
+			},
+			"name": {
+				Description: "The name of the personal access token.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"scopes": {
+				Description: fmt.Sprintf("The scope for the personal access token. It determines the actions which can be performed when authenticating with this token. Valid values are: %s.", renderValueListForDocs(validPersonalAccessTokenScopes)),
+				Type:        schema.TypeSet,
+				Required:    true,
+				ForceNew:    true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.StringInSlice(validPersonalAccessTokenScopes, false),
+				},
+			},
+			"active": {
+				Description: "True if the token is active.",
+				Type:        schema.TypeBool,
+				Computed:    true,
+			},
+			"revoked": {
+				Description: "True if the token is revoked.",
+				Type:        schema.TypeBool,
+				Computed:    true,
+			},
+			"created_at": {
+				Description: "Time the token has been created, RFC3339 format.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"expires_at": {
+				Description:      "The token expires at midnight UTC on that date. The date must be in the format YYYY-MM-DD. Default is never.",
+				Type:             schema.TypeString,
+				Optional:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: isISO6801Date,
+			},
+			"token": {
+				Description: "The personal access token. This is only populated when creating a new personal access token. This attribute is not available for imported resources.",
+				Type:        schema.TypeString,
+				Computed:    true,
+				Sensitive:   true,
+			},
+		},
+	}
+})
+
+func resourceGitlabPersonalAccessTokenCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+
+	userId := d.Get("user_id").(int)
```

Comment:
NIT: I suggest to get this attribute as close to where it's first used as possible ...

Reply:
Also, let's use `userID` instead of `userId` ...

Reply:
Implemented